tt_eager/tt_dnn/op_library/moreh_matmul/moreh_matmul_op.cpp

```diff
@@ -147,17 +170,20 @@ Tensor moreh_matmul_(
         uint32_t other_start_tile_id = other_other1_index * other_other2KtNt / TILE_HW;
         uint32_t output_start_tile_id = b1 * output_other2MtNt / TILE_HW;
 
-        operation::run(
+        output_tensor = operation::run(
```

Comment:
Notice here that we replace the original output_tensor with the one that was first created.   The reason for this is that it is a composite operation that runs multiple times against the same output tensor.